src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -932,6 +932,173 @@ public static void GetFullPath_Windows_Wildcards(char wildcard)
             AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetFullPath("test" + wildcard + "ing"));
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void GetFullPath_BasePath_InvalidInput()
+        {
+            Assert.Throws<ArgumentNullException>(() => Path.GetFullPath("", null));
+            Assert.Throws<ArgumentNullException>(() => Path.GetFullPath("tmp", null));
+            Assert.Throws<ArgumentNullException>(() => Path.GetFullPath(@"c:\git\corefx", null));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath("", @"foo\bar"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath("tmp", @"foo\bar"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C:\git", @"foo\bar"));
```

Comment:
For all of these argument exceptions, it'd be good to validate the parameter name as well (which will also mean switching from Assert.Throws to AssertExtensions.Throws).  Same for any other such asserts in the PR.